File: Domashno_2/guessable-api/src/main/kotlin/mk/ukim/finki/mis/guessableapi/domain/Location.kt
package mk.ukim.finki.mis.guessableapi.domain

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.*

/**
 * Represents a guessable location
 *
 * @property[id] unique identifier for the location
 * @property[latitude] the map latitude of the location
 * @property[longitude] the map longitude of the location
 * @property[image] [ByteArray] of the taken image for the location
 * @property[createdBy] the user that created the guessable location
 * */
@Entity
@Table(name = "locations")
data class Location(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long,

    @Column(name = "latitude")
    val latitude: Double,

    @Column(name = "longitude")
    val longitude: Double,

    @JsonIgnore
    @Column(name = "image")
    val image: ByteArray,

    @ManyToOne
    @JoinColumn(name = "created_by")
    val createdBy: GuessableUser,
)

File: Domashno_2/guessable-flutter-client/guessable/lib/domain/location.dart
import 'package:equatable/equatable.dart';
import 'package:guessable/domain/user.dart';

/// Represents a guessable location for the application
///
/// * [id] - the unique identifier for the [Location]
/// * [latitude] - the map latitude of the [Location]
/// * [longitude] - the map longitude of the [Location]
/// * [addedBy] - the [User] that created the [Location]
/// * [Location.fromJson] - used for creating a [Location] object from JSON, typically from HTTP requests
class Location extends Equatable {
  final int id;
  final double latitude;
  final double longitude;
  final User addedBy;

  const Location(
      {required this.id, required this.latitude, required this.longitude, required this.addedBy});

  factory Location.fromJson(Map<String, dynamic> json) {
    return Location(
        id: json['id'],
        latitude: json['latitude'],
        longitude: json['longitude'],
        addedBy: User.fromJson(json['createdBy']));
  }

  @override
  List<Object?> get props => [id, latitude, longitude, addedBy];
}
